core/types/digest.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2019 ChainSafe Systems (ON) Corp.
+// This file is part of gossamer.
+//
+// The gossamer library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The gossamer library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the gossamer library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"github.com/ChainSafe/gossamer/common"
+)
+
+// ConsensusEngineId is a 4-character identifier of the consensus engine that produced the digest.
+type ConsensusEngineId [4]byte
+
+var BabeEngineId = ConsensusEngineId{'B', 'A', 'B', 'E'}
+
+var ChangesTrieRootDigestType = byte(0)
+var PreRuntimeDigestType = byte(1)
+var ConsensusDigestType = byte(2)
+var SealDigestType = byte(4)
+
+// DigestItem can be of one of four types of digest: ChangesTrieRootDigest, PreRuntimeDigest, ConsensusDigest, or SealDigest.
+// see https://github.com/paritytech/substrate/blob/f548309478da3935f72567c2abc2eceec3978e9f/primitives/runtime/src/generic/digest.rs#L77
+type DigestItem interface {
+	Type() byte
+	Encode() []byte
+	//Decode([]byte)
```

Comment:
What's blocking this? Do you have plan to implement?

Reply:
nothing, I just didn't need it for this PR. will do in subsequent verification PRs

Reply:
Please open an issue to track (unless it's already WIP...)

Reply:
okay https://github.com/ChainSafe/gossamer/issues/523